File: kotlin/src/main/kotlin/com/lakamsani/farefinder/Flight.kt
package com.lakamsani.farefinder

/**
 * Represents a single flight option.
 */
data class Flight(
    val airline: String,
    val flightNumber: String,
    val price: Int,
    val departureTime: String,
    val arrivalTime: String,
    val duration: String,
)

File: flight.go
package main

import "fmt"

// Flight represents a single flight option.
type Flight struct {
	Airline       string
	FlightNumber  string
	Price         int
	DepartureTime string
	ArrivalTime   string
	Duration      string
}

func (f Flight) String() string {
	return fmt.Sprintf("Flight{airline='%s', flightNumber='%s', price=%d, departureTime='%s', arrivalTime='%s', duration='%s'}",
		f.Airline, f.FlightNumber, f.Price, f.DepartureTime, f.ArrivalTime, f.Duration)
}
